picas/actors.py

```diff
@@ -55,13 +57,18 @@ def _run(self, task, timeout):
         self.current_task = task
 
         try:
-            self.process_task(task, timeout=timeout)
+            with ThreadingTimeout(timeout, swallow_exc=False) as context_manager:
```

Comment:
Is there a reason to use `with` over a decorator? This style makes the code harder to read/maintain

Reply:
Using the context manager, allows catching the Timeout exception. Also, for some reason the decorator didn't work properly as it always returned None (while this should only be the case of timeout).